sql-plugin/src/main/330until340/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala

```diff
@@ -35,8 +34,9 @@ case class GpuBatchScanExec(
     output: Seq[AttributeReference],
     @transient scan: Scan,
     runtimeFilters: Seq[Expression] = Seq.empty,
-    keyGroupedPartitioning: Option[Seq[Expression]] = None)
-    extends ShimDataSourceV2ScanExecBase with GpuBatchScanExecMetrics {
+    keyGroupedPartitioning: Option[Seq[Expression]] = None,
+    ordering: Option[Seq[SortOrder]] = None)
```

Comment:
Spark 3.3 does not have this parameter, and it's unused here.

Reply:
This is required for 330db shim. Earlier it was working for 330db since we extended from `ShimDataSourceV2ScanExecBase` instead of `DataSourceV2ScanExecBase` and `ordering` was handled here - https://github.com/NVIDIA/spark-rapids/blob/branch-23.04/sql-plugin/src/main/311%2B-db/scala/com/nvidia/spark/rapids/shims/ShimLeafExecNode.scala#L37.

Please let me know what is preferred in this case :
1. Keep it as it is and add comment that it is used in 330db.
2. Modify code to use  ShimDataSourceV2ScanExecBase as before but only for 330until340 shim.
3. Create a separate GpuBatchScanExec for 311+-db shims

Reply:
> Keep it as it is and add comment that it is used in 330db.

I don't think this is an option, since I don't see where we're copying this parameter over in the 330db shim case.  Sure, it will compile on 330db, but if there was a non-None parameter for ordering, I don't see how the shim code properly copies this parameter over to the GPU instance.  Minimally we would need a 330db shim for BatchScanExecMeta to copy the ordering parameter.  Are there other BatchScanExec parameters on 330db that are not being handled properly?

> Modify code to use ShimDataSourceV2ScanExecBase as before but only for 330until340 shim

This seems reasonable if ordering is the only difference between the 330 and 330db versions (besides the BatchScanExecMeta change, of course).

> Create a separate GpuBatchScanExec for 311+-db shims

Assuming you meant 330db shim here, that's also an option.  Makes sense if there's more than ordering to take care of.

Reply:
There are additional parameters in DB-11.3, creating another version of GpuBatchScanExec in spark330db shim. We need to support `reusesFileListingResultsSourceNode`  on GPU. For now, we will fallback to CPU if `reusesFileListingResultsSourceNode !=None`

Reply:
Created follow on issue - https://github.com/NVIDIA/spark-rapids/issues/7750